src/Symfony/Component/DependencyInjection/Extension/CompilerPassProvidingExtensionInterface.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Symfony\Component\DependencyInjection\Extension;
+
+interface CompilerPassProvidingExtensionInterface
+{
+    /**
+     * @return array<CompilerPassInterface>|array<array{0: CompilerPassInterface, 1: string, 2?: int}>
```

Comment:
The right type is actually `array<CompilerPassInterface|array{0: CompilerPassInterface, 1: string, 2?: int}>` as you allow mixing both types of items in the returned array.

And you are missing the use statement for CompilerPassInterface

Reply:
I would also document the index 1 as `PassConfig::TYPE_*` instead of `string`, as we don't allow any string but only one of those constants (don't forget to add the usage statement)